sgv2-service-common/src/main/java/io/stargate/sgv2/common/metrics/ApiTimingDiagnostics.java

```diff
@@ -0,0 +1,71 @@
+package io.stargate.sgv2.common.metrics;
+
+import com.codahale.metrics.Timer;
+import java.util.function.Supplier;
+
+public class ApiTimingDiagnostics {
```

Comment:
This is extremely annoying, we are adding another level of complexity here for collecting the metrics.. Not sure if this should be merged in this way.. 

Simple timing management could be done easily later on with [@Counted and @Timed annotations](https://quarkus.io/guides/micrometer#does-micrometer-support-annotations). For the more complicated situations, we should use what's available in Micrometer, and function based time measurement is available: https://micrometer.io/docs/concepts#_timers

Reply:
I am not super happy about intrusiveness, but at the same time I don't think annotation based approaches can easily work in providing per-call tracing style metrics. They are fine for aggregates over time, but I am also interested in specific calls.
One benefit for explicit calls is that it is pretty clear what is happening, instead of something magical that happens somewhere else.

Reply:
Ok, so on Timers; this is about identical to DWM `Timer` so no objections to that part of handling (DWM->MM), as long as we can integrate MM metrics registry. At the same time use of these Timers is encapsulated within `ApiTimingDiagnostics` so it is also quite simple to change at a later point if need be.

But. This only addresses the aspect of exposing these via aggregate statistics over all calls. Like I said that is useful and all but does not help with tracing and collecting metrics for a single operation.
Passing callbacks (which I have started with) is certainly intrusive; an alternative by (ab)using `ThreadLocal`s would be ugly for other reasons.

Next thing I will be working on, and intended last part, is to make sampling of diagnostics to output configurable; disabled by default, and enabled only during testing.

Reply:
Added sampling/sampler, and also did convert to use Micrometer instead of dropwizard-metrics. Metrics reporting is bit different and looks like we actually lose functionality by doing this (we can only calculate average and do not get percentiles? (tp95, tp99 etc)) which is unfortunate.
But since we are moving to MM anyway I guess it makes sense, and for some usage it's fine.
Would be good to know if there was a way to get percentiles